linalg/hypre.cpp

```diff
@@ -2780,11 +2780,21 @@ void HypreParMatrix::PrintHash(std::ostream &os) const
 real_t HypreParMatrix::FNorm() const
 {
    real_t norm_fro = 0.0;
-   if (A != NULL)
+if (A != NULL)
+#if MFEM_HYPRE_VERSION >= 21900
    {
       const int ierr = hypre_ParCSRMatrixNormFro(A, &norm_fro);
       MFEM_VERIFY(ierr == 0, "");
    }
+#else
+   {   
+      MPI_Comm comm = hypre_ParCSRMatrixComm(A);
+      hypre_CSRMatrix *hypre_merged = hypre_MergeDiagAndOffd(A);
```

Comment:
Merging the diagonal and off-diagonal blocks is not a cheap operation. It is better to compute the norms of the blocks separately and then combine them.